jwst/barshadow/barshadow_step.py

```diff
@@ -0,0 +1,50 @@
+from ..stpipe import Step
+from .. import datamodels
+from . import bar_shadow
+
+
+class BarShadowStep(Step):
+    """
+    BarShadowStep: Inserts the bar shadow and wavelength arrays
+    into the data.
+
+    Bar shadow correction depends on the position of a pixel along the slit
+    and the wavelength. It is only applied to uniform sources and only for
+    NRS MSA data. 
+
+    """
+
+    spec = """
+    """
+
+    reference_file_types = ['barshadow']
+
+    def process(self, input):
+
+        # Open the input data model
+        with datamodels.open(input) as input_model:
+
+            # Get the name of the bar shadow reference file to use
+            self.barshadow_name = self.get_reference_file(input_model,
+                                                         'barshadow')
+            self.log.info('Using BARSHADOW reference file %s',
+                          self.barshadow_name)
+
+            # Check for a valid reference file
+            if self.barshadow_name == 'N/A':
+                self.log.warning('No BARSHADOW reference file found')
+                self.log.warning('Bar shadow step will be skipped')
+                result = input_model.copy()
+                result.meta.cal_step.barshadow = 'SKIPPED'
+                return result
+
+            instrument = input_model.meta.instrument.name
+            # Open the barshadow ref file data model
+            barshadow_model = datamodels.BarshadowModel(self.barshadow_name)
+
+            # Do the bar shadow correction     
+            result = bar_shadow.do_correction(input_model, barshadow_model)
+
+            barshadow_model.close()
+
+        return result
```

Comment:
You might want to add at the end of this step something like
```
if __name__ == '__main__':
    cmdline.step_script(BarShadowStep)
```
if this step can be called from the command line.

@hbushouse , should the cmdline be on all our steps?  I see it is on all our pipelines and many of our steps, but not all.

Reply:
This was a practice started many years ago in the early days of step development, largely before stpipe/strun was in use and hence allowed for the step to be executed directly from the command-line. It's really of little use anymore and I'm not aware of anyone who makes any regular use of it, so it's not required at all (IMHO).

Reply:
Ah, ok.  Right now 29 of our steps use this `cmdline.step_script` call.  Maybe we should do some purging?